src/bootstrap/compile.rs

```diff
@@ -1411,6 +1413,7 @@ pub fn run_cargo(
     stamp: &Path,
     additional_target_deps: Vec<(PathBuf, DependencyType)>,
     is_check: bool,
+    rlib_only_metadata: bool,
```

Comment:
This getting to be quite a lot of arguments. AFAICT `Cargo` is only used here and for `stream_cargo` - can we make `rlib_only_metadata` part of the `Cargo` builder instead of adding a new arg?

Reply:
`Cargo` represents a specific cargo binary together with the global cargo and rust flags for the current stage. Maybe combining `is_check` and `rlib_only_metadata` into a `CargoBuildMode` enum would work?

Reply:
hmm, ok. let's put this refactor off for later, it's not important for now.